src/pint/models/chromatic_model.py

```diff
@@ -0,0 +1,383 @@
+
+        return (
+            DMconst
+            * cm
+            * (bfreq / u.MHz) ** (-alpha)
+            * np.log((bfreq / u.MHz).to("").value)
+            * d_alpha_d_alphaparam
+        )
+
+    def register_cm_deriv_funcs(self, func, param):
+        """Register the derivative function in to the deriv_func dictionaries.
+
+        Parameters
+        ----------
+        func : callable
+            Calculates the derivative
+        param : str
+            Name of parameter the derivative is with respect to
+
+        """
+        pn = self.match_param_aliases(param)
+
+        if pn not in list(self.cm_deriv_funcs.keys()):
+            self.cm_deriv_funcs[pn] = [func]
+        elif func in self.cm_deriv_funcs[pn]:
+            return
+        else:
+            self.cm_deriv_funcs[pn] += [func]
+
+    def register_alpha_deriv_funcs(self, func, param):
+        """Register the derivative function in to the deriv_func dictionaries.
+
+        Parameters
+        ----------
+        func : callable
+            Calculates the derivative
+        param : str
+            Name of parameter the derivative is with respect to
+
+        """
+        pn = self.match_param_aliases(param)
+
+        if pn not in list(self.alpha_deriv_funcs.keys()):
+            self.alpha_deriv_funcs[pn] = [func]
+        elif func in self.alpha_deriv_funcs[pn]:
+            return
+        else:
+            self.alpha_deriv_funcs[pn] += [func]
+
+
+class ChromaticCM(Chromatic):
+    """Simple chromatic delay model.
+
+    This model uses Taylor expansion to model CM variation over time. It
+    can also be used for a constant CM.
+
+    Parameters supported:
+
+    .. paramtable::
+        :class: pint.models.chromatic_model.ChromaticCM
+    """
+
+    register = True
+    category = "chromatic_constant"
+
+    def __init__(self):
+        super().__init__()
+        self.add_param(
+            floatParameter(
+                name="CM",
+                units=cmu,
+                value=0.0,
+                description="Chromatic measure",
+                long_double=True,
+            )
+        )
+        self.add_param(
+            prefixParameter(
+                name="CM1",
+                units=cmu / u.year,
+                description="First order time derivative of the chromatic measure",
+                unit_template=self.CM_derivative_unit,
+                description_template=self.CM_derivative_description,
+                type_match="float",
+                long_double=True,
+            )
+        )
+        self.add_param(
+            MJDParameter(
+                name="CMEPOCH", description="Epoch of CM measurement", time_scale="tdb"
+            )
+        )
+        self.add_param(
+            floatParameter(
+                name="CMIDX",
+                units=u.dimensionless_unscaled,
+                value=4.0,
+                description="Chromatic measure index",
```

Comment:
Should this be "spectral index"?

Reply:
In PTA literature "spectral" is often used to describe the dependence on the Fourier conjugate frequency of TOAs (i.e, red noise frequencies, GW frequencies, etc). On the other hand, radio frequency dependence is often described as "chromatic". This is why I am using the term "chromatic" here.

Reply:
I was suggesting "Chromatic measure spectral index". But maybe that is too confusing.